File: backend/src/main/java/com/decmoe47/todo/service/TodoListService.java
package com.decmoe47.todo.service;

import com.decmoe47.todo.model.dto.TodoListAddDTO;
import com.decmoe47.todo.model.dto.TodoListDeleteDTO;
import com.decmoe47.todo.model.dto.TodoListUpdateDTO;
import com.decmoe47.todo.model.vo.TodoListVO;

import java.util.List;

public interface TodoListService {

    List<TodoListVO> getCustomTodoLists(long userId);

    TodoListVO addTodoList(TodoListAddDTO todoListAddDTO);

    TodoListVO updateTodoList(TodoListUpdateDTO todoListUpdateDTO);

    void deleteTodoList(TodoListDeleteDTO todoListDeleteDTO);
}

File: backend/src/main/kotlin/com/decmoe47/todo/service/TodoListService.kt
package com.decmoe47.todo.service

import com.decmoe47.todo.model.request.TodoListAddRequest
import com.decmoe47.todo.model.request.TodoListDeleteRequest
import com.decmoe47.todo.model.request.TodoListUpdateRequest
import com.decmoe47.todo.model.response.TodoListResponse

interface TodoListService {
    fun getAll(): List<TodoListResponse>

    fun add(request: TodoListAddRequest): TodoListResponse

    fun update(request: TodoListUpdateRequest): TodoListResponse

    fun delete(request: TodoListDeleteRequest)
}
